crates/db_views/src/custom_emoji_view.rs

```diff
@@ -0,0 +1,86 @@
+use crate::structs::CustomEmojiView;
+use diesel::{
+  result::Error,
+  BelongingToDsl,
+  ExpressionMethods,
+  JoinOnDsl,
+  NullableExpressionMethods,
+  QueryDsl,
+};
+use diesel_async::RunQueryDsl;
+use lemmy_db_schema::{
+  newtypes::{CustomEmojiId, LocalSiteId},
+  schema::{custom_emoji, custom_emoji_keyword},
+  source::{custom_emoji::CustomEmoji, custom_emoji_keyword::CustomEmojiKeyword},
+  traits::ViewToVec,
+  utils::{get_conn, DbPool},
+};
+use std::collections::HashMap;
+
+type CustomEmojiTuple = (CustomEmoji, Option<CustomEmojiKeyword>);
+
+impl CustomEmojiView {
+  pub async fn get(pool: &DbPool, emoji_id: CustomEmojiId) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    let emoji = custom_emoji::table
+      .find(emoji_id)
+      .first::<CustomEmoji>(conn)
+      .await?;
+    let keywords = CustomEmojiKeyword::belonging_to(&emoji)
+      .load::<CustomEmojiKeyword>(conn)
+      .await?;
+
+    let view = CustomEmojiView {
+      custom_emoji: emoji,
+      keywords,
+    };
+
+    Ok(view)
+  }
+
+  pub async fn get_all(pool: &DbPool, for_local_site_id: LocalSiteId) -> Result<Vec<Self>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    let emojis = custom_emoji::table
+      .filter(custom_emoji::local_site_id.eq(for_local_site_id))
+      .left_join(
+        custom_emoji_keyword::table.on(custom_emoji_keyword::custom_emoji_id.eq(custom_emoji::id)),
+      )
+      .order(custom_emoji::category)
+      .then_order_by(custom_emoji::id)
+      .select((
+        custom_emoji::all_columns,
+        custom_emoji_keyword::all_columns.nullable(), // (or all the columns if you want)
+      ))
+      .load::<CustomEmojiTuple>(conn)
+      .await?;
+
+    Ok(CustomEmojiView::from_tuple_to_vec(emojis))
+  }
+}
+
+impl ViewToVec for CustomEmojiView {
+  type DbTuple = CustomEmojiTuple;
+  fn from_tuple_to_vec(items: Vec<Self::DbTuple>) -> Vec<Self> {
+    let mut result = Vec::new();
+    let mut hash: HashMap<CustomEmojiId, Vec<CustomEmojiKeyword>> = HashMap::new();
+    for item in &items {
+      let emoji_id: CustomEmojiId = item.0.id;
+      if let std::collections::hash_map::Entry::Vacant(e) = hash.entry(emoji_id) {
+        e.insert(Vec::new());
+        result.push(CustomEmojiView {
+          custom_emoji: item.0.clone(),
+          keywords: Vec::new(),
+        })
+      }
+      if item.1.is_some() {
+        if let Some(keywords) = hash.get_mut(&emoji_id) {
+          keywords.push(item.1.as_ref().unwrap().clone())
+        }
+      }
+    }
+    for emoji in &mut result {
+      emoji.keywords = hash.get(&emoji.custom_emoji.id).unwrap().clone();
+    }
+    result
```

Comment:
This to me is very similar to what `array_agg` or `string_agg` would do, except that its taking smart processing away from postgres and moving it to code. Hopefully this doesn't add too much CPU processing overhead.

Reply:
I can't imagine what diesel would do with a array_agg... or how to map it properly. No docs i could find on handling it. Either way, this is pretty efficient for what it does.

Reply:
There's a good chance diesel can't do array_agg or string_agg, oh well. 

As long as this works, and especially since its not chaining queries, we should be okay.